mlflow/transformers.py

```diff
@@ -1396,13 +1398,41 @@ def predict(self, data):
                 for x in input_data
             )
 
-        predictions = self._predict(input_data)
+        predictions = self._predict(input_data, device)
 
         return predictions
 
-    def _predict(self, data):
+    # This function attempts to determine if a GPU is available for the PyTorch and TensorFlow libraries
+    def is_gpu_available(self):
+        # try pytorch and if it fails, try tf
+        is_gpu = None
+        try:
+            import torch
+            is_gpu = torch.cuda.is_available()
+        except ImportError:
+            pass
+        if is_gpu is None:
+            try:
+                import tensorflow as tf
+                is_gpu = tf.test.is_gpu_available()
+            except ImportError:
+                pass
+        if is_gpu is None:
+            is_gpu = False
+        return is_gpu
+
+
+    def _predict(self, data, device):
         import transformers
-
+        if device is None:
+            if MLFLOW_DEFAULT_PREDICTION_DEVICE.get():
+                device = MLFLOW_DEFAULT_PREDICTION_DEVICE.get()
+            elif self.is_gpu_available():
+                device = _TRANSFORMERS_DEFAULT_GPU_DEVICE_ID
+            else:
+                device = None
+        if device:
+            self.inference_config['device'] = device
```

Comment:
Isn't 0, which is the value of `_TRANSFORMERS_DEFAULT_GPU_DEVICE_ID`, a falsey value? If so, this doesn't seem to work :).

It would be awesome if we could write a unit test that mocks the call to `self.pipeline()` and verifies that the expected inference config is being passed in.

Reply:
```suggestion
        if device is not None:
            self.inference_config['device'] = device
```

Reply:
Agreed and modified to adjust to this in latest commit!